ballista/rust/scheduler/src/state/task_manager.rs

```diff
@@ -37,16 +37,19 @@ use ballista_core::serde::{AsExecutionPlan, BallistaCodec};
 use datafusion::physical_plan::ExecutionPlan;
 use datafusion::prelude::SessionContext;
 use datafusion_proto::logical_plan::AsLogicalPlan;
-use log::{debug, info, warn};
+#[cfg(not(test))]
+use log::info;
+use log::{debug, error, warn};
 use rand::distributions::Alphanumeric;
 use rand::{thread_rng, Rng};
-use std::collections::{HashMap, HashSet};
+use std::collections::HashMap;
 use std::default::Default;
 use std::sync::Arc;
 use tokio::sync::RwLock;
 use tonic::transport::Channel;
 
 type ExecutorClients = Arc<RwLock<HashMap<String, ExecutorGrpcClient<Channel>>>>;
+type ExecutionGraphCache = Arc<RwLock<HashMap<String, Arc<RwLock<ExecutionGraph>>>>>;
```

Comment:
What do you think about making this a `Arc<RwLock<VecDequeue<(String,ExecutionGraph)>>>`? Lookup by job ID becomes `O(n)` but the number of active jobs shouldn't be prohibitive and you get a proper queue in return so we can do FIFO scheduling of new jobs.

Reply:
I don't think it's a good idea 🤣. If we dequeue the `ExecutionGraph`, and meanwhile we lookup the graph, we can get nothing. And the number of active jobs may be large. In our case, it may be thousands.

Reply:
Ok, we can sort it out in a future PR. But I think we do want to take advantage of this change to introduce proper queueing for the jobs for fair scheduling. It could be as simple as another `VecDequeue<String>` that holds represents the queue which we could use to indicate priority.